pkg/scalers/aws/aws_authorization.go

```diff
@@ -0,0 +1,17 @@
+package aws
+
+type AuthorizationMetadata struct {
+	AwsRoleArn string
+
+	AwsAccessKeyID     string
+	AwsSecretAccessKey string
+	AwsSessionToken    string
+
+	PodIdentityOwner bool
+	// Pod identity owner is confusing
+	// and it'll be removed when we get
+	// rid of the old aws podIdentities
+	UsingPodIdentity bool
```

Comment:
What is the difference between these two?

Reply:
`PodIdentityOwner` is confusing (at least from my pov) because it means that we are using podIdentity but only using KEDA's identity. `UsingPodIdentity` means that we are using podIdentity, and if we use KEDA's role or other is known by other properties.
Why do we need to know this? Because it's not the same code for using static credentias, KEDA's role or workload/random role, and this allows use to do something like:
```go
	if awsAuthorization.UsingPodIdentity {
		if awsAuthorization.AwsRoleArn != "" {
			cfg.Credentials = a.retrievePodIdentityCredentials(cfg, awsAuthorization.AwsRoleArn)
		}
	} else {
		cfg.Credentials = a.retrieveStaticCredentials(awsAuthorization)
	}
```
If we use podIdentity, we don't want to use static credentials, but we still have to decide if we want to use self or other role, based on AwsRoleArn. 
I see it more clear (but I'm open to discuss if we should keep the old one). Another point for the new one, is that as we are deprecating the old auths, we can easily distinguish between the new one and the old one (because they are totally different and use different vars)

Reply:
Okay, would be great to put this context into the source code!

Reply:
isn't too much text? I mean, without any context I've just said that we have to remove it and when xD

Reply:
Let me add a link to this comment, if someone want more context, they can come here and check it

Reply:
done